linkerd/proxy/server-policy/src/lib.rs

```diff
@@ -131,6 +135,28 @@ pub mod proto {
                 server_ips: _,
             } = proto;
 
+            let local_rate_limit = {
+                match protocol
+                    .clone()
+                    .and_then(|api::ProxyProtocol { kind }| kind)
+                    .ok_or(InvalidServer::MissingProxyProtocol)?
+                {
+                    api::proxy_protocol::Kind::Detect(api::proxy_protocol::Detect {
+                        http_local_rate_limit,
+                        ..
+                    }) => http_local_rate_limit.unwrap_or_default().into(),
+                    api::proxy_protocol::Kind::Http1(api::proxy_protocol::Http1 {
+                        local_rate_limit,
+                        ..
+                    })
+                    | api::proxy_protocol::Kind::Http2(api::proxy_protocol::Http2 {
+                        local_rate_limit,
+                        ..
+                    }) => local_rate_limit.unwrap_or_default().into(),
+                    _ => Default::default(),
+                }
+            };
```

Comment:
nit: We can make this slightly more compact by removing redundant braces. The other blocks that do this define temporary variables.

```suggestion
            let local_rate_limit = match protocol
                .clone()
                .and_then(|api::ProxyProtocol { kind }| kind)
                .ok_or(InvalidServer::MissingProxyProtocol)?
            {
                api::proxy_protocol::Kind::Detect(api::proxy_protocol::Detect {
                    http_local_rate_limit,
                    ..
                }) => http_local_rate_limit.unwrap_or_default().into(),
                api::proxy_protocol::Kind::Http1(api::proxy_protocol::Http1 {
                    local_rate_limit,
                    ..
                })
                | api::proxy_protocol::Kind::Http2(api::proxy_protocol::Http2 {
                    local_rate_limit,
                    ..
                }) => local_rate_limit.unwrap_or_default().into(),
                _ => Default::default(),
            };
```